### PR TITLE
goose: init at 3.11.2

### PR DESCRIPTION
**Pull request by yu-re-ka**
###### Description of changes

Tested functionality

<del>Can not figure out how to make `goose -version` output the correct thing</del>

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/goose/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    "-s"
+    "-w"
+    "-X main.gooseVersion=${version}"
+  ];
+
+  checkFlags = [ "-skip=${lib.concatStringsSep "|" [
+    # these also require a docker daemon
+    "TestClickUpDown"
+    "TestClickHouseFirstThree"
+  ]}" ];
```
Comment (figsoda) [inline pkgs/tools/misc/goose/default.nix:34]:
```suggestion
  checkFlags = [
    # these also require a docker daemon
    "-skip=TestClickUpDown|TestClickHouseFirstThree"
  ];
```

File: pkgs/tools/misc/goose/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  postPatch = ''
+    rm -r tests/e2e
+    rm -r tests/gomigrations
+    rm -r tests/vertica
+  '';
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.gooseVersion=${version}"
```
Comment (figsoda) [inline pkgs/tools/misc/goose/default.nix:27]:
```suggestion
    "-X=main.gooseVersion=${version}"
```

File: pkgs/tools/misc/goose/default.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  ];
+
+  checkFlags = [ "-skip=${lib.concatStringsSep "|" [
+    # these also require a docker daemon
+    "TestClickUpDown"
+    "TestClickHouseFirstThree"
+  ]}" ];
+
+  vendorHash = "sha256-n5aUV+FXjfAmzToRH/v4Rq9/sUkmVaSvo32058q/3nE=";
```
Comment (figsoda) [inline pkgs/tools/misc/goose/default.nix:36]: could you move this right below `src`?

**Review by figsoda — COMMENTED** [review]
just some nitpicks

File: pkgs/tools/misc/goose/default.nix (hunk 3 above)
Comment (yu-re-ka) [reply pkgs/tools/misc/goose/default.nix]: done

File: pkgs/tools/misc/goose/default.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  version = "3.11.2";
+
+  src = fetchFromGitHub {
+    owner = "pressly";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-D/mWzDsGNftl0ldNpWt/98b2o5fWk4Jxb5zEIAqTPs0=";
+  };
+
+  vendorHash = "sha256-n5aUV+FXjfAmzToRH/v4Rq9/sUkmVaSvo32058q/3nE=";
```
Comment (figsoda) [inline pkgs/tools/misc/goose/default.nix:17]:
```suggestion
  proxyVendor = true;
  vendorHash = "sha256-6KqYafXtTLmiYrKabuPaszzkN5P3uZDl4LWo1tat/Bk=";
```

File: pkgs/tools/misc/goose/default.nix [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    "-s"
+    "-w"
+    "-X=main.gooseVersion=${version}"
+  ];
+
+  checkFlags = [
+    # these also require a docker daemon
+    "-skip=TestClickUpDown|TestClickHouseFirstThree"
+  ];
+
```
Comment (figsoda) [inline pkgs/tools/misc/goose/default.nix:36]: I couldn't figure out why or what tests are failing on darwin

```suggestion

  doCheck = !stdenv.isDarwin;

```

**Review by figsoda — COMMENTED** [review]
this fixes the build on darwin

**Comment by yu-re-ka** [comment]
Thanks!

**Comment by figsoda** [comment]
oops stdenv needs to be added to the inputs

**Comment by yu-re-ka** [comment]
> oops stdenv needs to be added to the inputs

done

**Review by figsoda — APPROVED** [review]

**Comment by yu-re-ka** [comment]
@ofborg build goose

**Pull request merged**